HandleNHNotifyMessageCallActivityQueue/handler.ts

```diff
@@ -0,0 +1,95 @@
+import { flow, pipe } from "fp-ts/lib/function";
+import * as E from "fp-ts/Either";
+import * as TE from "fp-ts/TaskEither";
+import { FiscalCode } from "@pagopa/ts-commons/lib/strings";
+import { TelemetryClient } from "applicationinsights";
+import { errorsToError } from "../IsUserInActiveSubsetActivity/handler";
+import {
+  buildNHService,
+  getNotificationHubPartitionConfig,
+  NotificationHubConfig
+} from "../utils/notificationhubServicePartition";
+import { toSHA256 } from "../utils/conversions";
+import { notify } from "../utils/notification";
+import { NhNotifyMessageRequest } from "../utils/types";
+import {
+  Failure,
+  throwTransientFailure,
+  toPermanentFailure,
+  toTransientFailure
+} from "../utils/errors";
+
+export type NhNotifyMessageResponse = Promise<
+  Failure | { readonly kind: string }
+>;
+
+export const handle = (
+  inputRequest: unknown,
+  legacyNotificationHubConfig: NotificationHubConfig,
+  notificationHubConfigPartitionChooser: ReturnType<
+    typeof getNotificationHubPartitionConfig
+  >,
+  fiscalCodeNotificationBlacklist: ReadonlyArray<FiscalCode>,
+  telemetryClient: TelemetryClient
+): NhNotifyMessageResponse =>
+  pipe(
+    inputRequest,
+    NhNotifyMessageRequest.decode,
+    E.mapLeft(flow(errorsToError, toPermanentFailure)),
+    TE.fromEither,
+    TE.bindTo("request"),
+    TE.bind(
+      "nhConfig",
+      ({
+        request: {
+          message: { installationId },
+          target
+        }
+      }) =>
+        TE.of(
+          target === "legacy"
+            ? legacyNotificationHubConfig
+            : notificationHubConfigPartitionChooser(installationId)
+        )
+    ),
+    TE.chain(({ request: { message }, nhConfig }) =>
+      pipe(
+        { kind: "SUCCESS", skipped: true },
+        TE.fromPredicate(
+          () =>
+            fiscalCodeNotificationBlacklist
+              .map(toSHA256)
+              .includes(message.installationId),
+          () => "execute notify"
+        ),
+        TE.orElseW(() =>
+          pipe(
+            notify(
+              buildNHService(nhConfig),
+              message.installationId,
+              message.payload
+            ),
+            TE.map(() => ({ kind: "SUCCESS", skipped: false }))
+          )
+        ),
+        TE.mapLeft(toTransientFailure),
```

Comment:
Could be useful to also track an `api.messages.notification.push.sent` event in case of failure, with `isSuccess` set to `false`?

Reply:
added. thank you!